### DISCUSSION TITLE
RN Firebase x `recaptchaToken`

### DISCUSSION THREAD

**Discussion by kperreau** [invertase/react-native-firebase#6003] [Q&A]
Hi!

Is there a way to get the `recaptchaToken` from RN Firebase SDK ?
I mean, when you are using Firebase auth to verify a phone number, there is an hidden captcha, SafetyNet in Android for exemple.
And this is required to request the google API verification code.

I don't want to use the built in firebase.auth() client side anymore,  i want to send data to my own backend, and then from my backend i use this Google API:
```
curl -X POST \
  'https://www.googleapis.com/identitytoolkit/v3/relyingparty/sendVerificationCode?key=api_key' \
  -H 'content-type: application/json' \
  -d '{
 "phoneNumber": "phone_number_to_verify",
 "recaptchaToken": "generated_recaptcha_token"
}'
```

So, it would be great if the `recaptchaToken` is still generated rom RN Firebase, but i don't find a method to get it.

**Comment by mikehardy** [comment]
Sorry I don't have an answer for you - this is a first request on that one. I have some difficulty in answering this because when you say "RN Firebase SDK", my initial thought is "Firebase does not have an RN SDK", what we have in react-native is a "firebase SDK wrapper", which leads to the question, "okay, which SDK?". We wrap firebase-android-sdk and firebase-ios-sdk here, with an API that is as close to firebase-js-sdk v8 (not v9 yet) as possible. So which SDK do you mean?

I do not believe that either of the native (android or ios) firebase SDKs provide this, but please do check the upstream reference docs for those SDKs. firebase-js-sdk may, but I'm not sure on that, and we don't actually use firebase-js-sdk here so we couldn't provide it anyway, that would be your project just using firebase-js-sdk directly
